AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -411,7 +409,9 @@ - (void)setAsyncDelegate:(id<ASCollectionViewDelegate>)asyncDelegate
 
   super.delegate = (id<UICollectionViewDelegate>)_proxyDelegate;
   
-  [_layoutInspector didChangeCollectionViewDelegate:asyncDelegate];
+  if ([_layoutInspector respondsToSelector:@selector(didChangeCollectionViewDelegate:)]) {
```

Comment:
I'm not pretty sure if we should cache these calls as we have to be careful internally to always use the setter for the layout inspector and not the instance variable to wire up the flags. As I said the call should  not happen that often, frankly it will happen more often as someone could just set the delegate and if we not cache it we call only the delegate `respondToSelector:` call and not the data source one. If we cache it we always make it twice to cache it.